src/Cogworks.CogFlare.Core/Services/CachePurgeService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using static Umbraco.Cms.Core.Constants.Conventions;
+
+namespace Cogworks.CogFlare.Core.Services;
+
+public interface ICachePurgeService
+{
+    Task PurgeExternalCacheAsync(IEnumerable<int> ids, CancellationToken cancellationToken, string notificationLabel, bool isMedia = false);
+}
+
+public class CachePurgeService : ICachePurgeService
+{
+    private readonly ICloudFlareCachePurgeService _cloudFlareCachePurgeService;
+    private readonly IUmbracoContentNodeService _umbracoContentNodeService;
+    private readonly IRelationService _relationService;
+    private readonly CogFlareSettings _cogFlareSettings;
+    private readonly ILogger<ExternalCachePurgeComponent> _logger;
+
+    public CachePurgeService(
+        ICloudFlareCachePurgeService cloudFlareCachePurgeService,
+        IUmbracoContentNodeService umbracoContentNodeService,
+        IRelationService relationService,
+        CogFlareSettings cogFlareSettings,
+        ILogger<ExternalCachePurgeComponent> logger)
+    {
+        _cloudFlareCachePurgeService = cloudFlareCachePurgeService;
+        _umbracoContentNodeService = umbracoContentNodeService;
+        _relationService = relationService;
+        _cogFlareSettings = cogFlareSettings;
+        _logger = logger;
+    }
+
+
+    public async Task PurgeExternalCacheAsync(
+        IEnumerable<int> ids,
+        CancellationToken cancellationToken,
+        string notificationLabel,
+        bool isMedia = false)
+    {
+        var urlsToPurge = new List<string>();
+
+        foreach (var id in ids)
+        {
+            if (IsKeyNode(id))
+            {
+                _logger.LogInformation($"Full purge triggered: [{id}] Key node {notificationLabel}");
```

Comment:
Are we sure we always want to log this information? Do we ever use this?

Reply:
I think it's required to have knowledge of how many times the user make a purge

Reply:
Yes. That comes in very handy for debugging. I plan on making the logging optional with a config setting or feature flag in a later update

Reply:
Yep so it's only helpful when debugging. Once on production it produces too many longs so Umbraco stops showing ALL logs in the Log Viewer. I'd say introduce the config setting now, it's a 5 min job.